### ISSUE TITLE
[BUG]: TypeScript compilation errors with @octokit/rest and latest version of @octokit-types

### ISSUE THREAD

**Issue by bcoe** [octokit/rest.js#290]
### What happened?

TypeScript code no longer compiles with the following errors:

```
node_modules/@octokit/plugin-rest-endpoint-methods/dist-types/generated/parameters-and-response-types.d.ts:3091:60 - error TS2339: Property 'POST /enterprises/{enterprise}/{security_product}/{enablement}' does not exist on type 'Endpoints'.

3091             parameters: RequestParameters & Omit<Endpoints["POST /enterprises/{enterprise}/{security_product}/{enablement}"]["parameters"], "baseUrl" | "headers" | "mediaType">;
                                                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@octokit/plugin-rest-endpoint-methods/dist-types/generated/parameters-and-response-types.d.ts:3092:33 - error TS2339: Property 'POST /enterprises/{enterprise}/{security_product}/{enablement}' does not exist on type 'Endpoints'.

3092             response: Endpoints["POST /enterprises/{enterprise}/{security_product}/{enablement}"]["response"];
```

### Versions

`>v19`.

### Relevant log output

```shell
node_modules/@octokit/plugin-rest-endpoint-methods/dist-types/generated/parameters-and-response-types.d.ts:3091:60 - error TS2339: Property 'POST /enterprises/{enterprise}/{security_product}/{enablement}' does not exist on type 'Endpoints'.

3091             parameters: RequestParameters & Omit<Endpoints["POST /enterprises/{enterprise}/{security_product}/{enablement}"]["parameters"], "baseUrl" | "headers" | "mediaType">;
                                                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@octokit/plugin-rest-endpoint-methods/dist-types/generated/parameters-and-response-types.d.ts:3092:33 - error TS2339: Property 'POST /enterprises/{enterprise}/{security_product}/{enablement}' does not exist on type 'Endpoints'.

3092             response: Endpoints["POST /enterprises/{enterprise}/{security_product}/{enablement}"]["response"];
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by wolfy1339** [comment]
This is a known issue and is being worked on.

The version of Octokit packages and the types packages are tied together.
You cannot upgrade one without the other.

There need to be an update to the `@octokit/plugin-rest-endpoint-methods`, `@octokit/plugin-throttling`, and `@octokit/types`

Currently blocked on https://github.com/octokit/types.ts/issues/534 which hopefully will get fixed by https://github.com/octokit/openapi/pull/349

**Comment by mattgi** [comment]
Is this the same issue as this?

![CleanShot 2023-05-08 at 20 31 56](https://user-images.githubusercontent.com/1043906/236802155-721561d5-2f3c-47ad-84ab-0b19afc9c762.png)

**Comment by coxd** [comment]
@mattgi 

It's a different set of errors but same underlying cause.

This is PR that will eventually solve all I believe: https://github.com/octokit/plugin-rest-endpoint-methods.js/pull/632

**Comment by wolfy1339** [comment]
Hopefully it will get resolved as soon as https://github.com/octokit/openapi/pull/353 is merged.

**Comment by Louis-7** [comment]
Is there a workaround for this issue, or any backward version I can use?

**Comment by Louis-7** [comment]
> Is there a workaround for this issue, or any backward version I can use?

Error has gone after I used this:
```
{
  "dependencies": {
    "@octokit/rest": "^18.12.0"
  },
  "devDependencies": {
    "@octokit/types": "^9.2.1",
    "typescript": "^5.0.4"
  }
}
```
The only thing I did was downgrade the `@octokit/rest`

**Comment by wolfy1339** [comment]
Yes

In this case it's `@octokit/plugin-paginate-rest` that is still waiting

**Comment by wolfy1339** [comment]
Every dependant module that relies upon the endpoint types has been updates, I'm now going to release a new version of `@octokit/rest` with those updated modules.

See #292

**Comment by josecosta-shell** [comment]
Still having this problem when using `"@octokit/rest": "^20.0.1"` and `"@octokit/types": "^11.1.0"`. Specifically using `octokit.repos.getContent`, I get a message saying the `data.sha` property doesn't exist:

```
Property 'sha' does not exist on type '{ type: "dir" | "file" | "submodule" | "symlink"; size: number; name: string; path: string; content?: string | undefined; sha: string; url: string; git_url: string | null; html_url: string | null; download_url: string | null; _links: { ...; }; }[] | { ...; } | { ...; } | { ...; }'.
  Property 'sha' does not exist on type '{ type: "dir" | "file" | "submodule" | "symlink"; size: number; name: string; path: string; content?: string | undefined; sha: string; url: string; git_url: string | null; html_url: string | null; download_url: string | null; _links: { ...; }; }[]'.ts(2339)
```

**Comment by wolfy1339** [comment]
Can you open a new issue, that is different than  this one.
